### PR TITLE
Eliminates reference to topic tree node children as anything but ids

### LINKED ISSUE #3743: Uncaught error on Learn page
On `develop`:
Observed behavior:
Occasionally (I haven't been able to reproduce this reliably yet :( ) I see the following error in a red bubble on the learn (`/learn/khan/`) page:
>Unexpected error: 'str' object has no attribute 'get'

Steps to reproduce:
Unknown at this time. Please report if you encounter this error, too!

### PR DESCRIPTION
**Pull request by rtibbles**
Fixes #3743 - except it doesn't. But fixes the only possible places I can find in the code that might be associated with this.

Summary of changes:
* Fully deprecates Playlists in student progress reports.
* Eliminates reference to topic tree node children as anything but ids.

@MCGallaspy

### CODE REVIEW

**Comment by MCGallaspy** [comment]
Cool cool cool. So did you figure out how to produce such an error? I was never able to pin it down.

File: kalite/coachreports/models.py [hunk 1]
```diff
@@ -267,31 +260,31 @@ def user_progress_detail(cls, user_id, playlist_id):
...
-
-                    entry = {
-                        "id": quiz_log_id,
-                        "kind": "Quiz",
-                        "status": status,
-                        "score": quiz_pct_score,
-                        "title": playlist.get("title"),
-                        "path": "",
-                    }
+            # elif kind == "Quiz":
```
Comment (MCGallaspy) [inline kalite/coachreports/models.py]: I'll take your word that these changes are okay, but can you either remove the commented out code or add some context as to why it remains there?

**Comment by rtibbles** [comment]
@MCGallaspy I never pinned it down, I just grepped the codebase for places where we loop over the children of a topic node, and made sure we only ever treated them as keys, not as objects.

**Comment by rtibbles** [comment]
@MCGallaspy This should be good to merge, methinks. Comments added.

**Pull request merged**